doc/release.md

```diff
@@ -2,46 +2,93 @@
 
 So, you want to release the `X.Y.Z` version of astroid ?
 
-## Process
+## Releasing a major or minor version
 
-(Consider triggering the "release tests" workflow in GitHub Actions first.)
+**Before releasing a major or minor version check if there are any unreleased commits on
+the maintenance branch. If so, release a last patch release first. See
+`Release a patch version`.**
 
-1. Check if the dependencies of the package are correct
-2. Check the result (Do `git diff vX.Y.Z-1 ChangeLog` in particular).
-3. Install the release dependencies `pip3 install -r requirements_test.txt`
-4. Bump the version and release by using `tbump X.Y.Z --no-push`.
-5. Push the tag.
-6. Release the version on GitHub with the same name as the tag and copy and paste the
+1. Remove the empty changelog for the last unreleased patch version `X.Y-1.Z'`. (For
+   example: `v2.3.5`)
+2. Check the result of `git diff vX.Y-1.Z' ChangeLog`. (For example:
+   `git diff v2.3.4 ChangeLog`)
+3. Install the release dependencies: `pip3 install -r requirements_test.txt`
+4. Bump the version and release by using `tbump X.Y.0 --no-push`. (For example:
+   `tbump 2.4.0 --no-push`)
+5. Check the result visually and then by triggering the "release tests" workflow in
+   GitHub Actions first.
+6. Push the tag.
+7. Release the version on GitHub with the same name as the tag and copy and paste the
    appropriate changelog in the description. This trigger the pypi release.
+8. Move the `main` branch up to a dev version with `tbump`:
 
-## Post release
-
-### Back to a dev version
+```bash
+tbump X.Y+1.0-dev0 --no-tag --no-push  # You can interrupt after the first step
+git commit -am "Upgrade the version to x.y+1.0-dev0 following x.y.0 release"
+```
 
-Move back to a dev version with `tbump`:
+For example:
 
 ```bash
-tbump X.Y.Z+1-dev0 --no-tag --no-push # You can interrupt after the first step
-git commit -am "Upgrade the version to x.y.z+1-dev0 following x.y.z release"
+tbump 2.5.0-dev0 --no-tag --no-push
+git commit -am "Upgrade the version to 2.5.0-dev0 following 2.4.0 release"
 ```
 
 Check the result and then upgrade the main branch
 
-### Milestone handling
+11. Delete the `maintenance/X.Y-1.x` branch. (For example: `maintenance/2.3.x`)
+
+## Backporting a fix from `main` to the maintenance branch
+
+Whenever a commit on `main` should be released in a patch release on the current
+maintenance branch we cherry-pick the commit from `main`.
+
+1. During the merge request on `main`, make sure that the changelog is for the patch
+   version `X.Y-1.Z'`. (For example: `v2.3.5`)
+2. Create a `maintenance/X.Y.x` branch on Github --if it does not already exist-- from
```

Comment:
I think it's good to make the `maintenance` branch always as it is also a visual pointer about which versions are still supported.

Reply:
[f594919](https://github.com/PyCQA/astroid/pull/1438/commits/f59491952604e9690699c74035b917e445025d09)